### PR TITLE
Feature/mds 3238 - Populate conditions with variables

### PR DESCRIPTION
**Pull request by mmumby**
# Main

- Ability to add variables to the permit conditions on the UI
- Ability to map those variables to data points on the NoW, permit, mine, etc and display on the permit pdf in place of the variables
- The "Permit Condition Management" view displays a default list of reclamation activities, The draft permit tab displays only the activities related to the NoW.

# Other
- Updated test coverage on all condition components. 
- Added a modal to view all permit conditions on the Permit table. 
Screen Shot 2021-10-15 at 5 04 02 PM
Screen Shot 2021-10-15 at 5 04 49 PM
Screen Shot 2021-10-15 at 5 14 02 PM

# How to test

- Create a draft permit -> alter the conditions to include some or all of the variables listed in the menu -> Click "download Draft" and confirm that the pdf is now displaying the correct data in place of the variable names. 
Test Cases 
1. when you click the edit pencil, a menu appears in the top left corner. 
2. Selecting an option from the menu adds a variable to the input box in this format: {variable}
3. When you click save on the permit condition, the variable gets highlighted yellow. 
4. When you click "download Draft Permit" it downloads successfully, the variable is either replaced with the data or an empty string if the data is null. 

Repeat steps 1-3 on the "Permit Condition Management" tab under the Admin menu.

# Notes
Screen Shot 2021-10-15 at 1 51 01 PM
Screen Shot 2021-10-15 at 1 51 20 PM
Screen Shot 2021-10-15 at 1 51 34 PM
Screen Shot 2021-10-15 at 1 51 53 PM
Screen Shot 2021-10-15 at 1 52 31 PM
Screen Shot 2021-10-15 at 1 53 01 PM
Screen Shot 2021-10-15 at 1 53 33 PM

### CODE REVIEW

File: services/core-web/src/components/Forms/permits/conditions/Conditions.js
Comment (mmumby) [inline services/core-web/src/components/Forms/permits/conditions/Conditions.js:199]: line 198 is the only addition, everything else in this file is formatting.

File: services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:540]: The "Download draft" throws an error from the BE if there is no signature, and since we are already disabling based on the existence of the permittee, I've added a disable check for a signature too so there is no longer a BE error.

**Review by dragancolic-bcgov — DISMISSED** [review]
From my point of view it looks good. Perhaps to be more confident in merging the code you may want to wait for one more approval. Great job Meredith!

**Review by dragancolic-bcgov — DISMISSED** [review]

File: services/core-api/app/api/now_applications/models/now_application_document_type.py
Comment (areyeslo) [inline services/core-api/app/api/now_applications/models/now_application_document_type.py]: I'd remove the for loop layers to iterate in different way through the levels of the conditions. Something like the following:

```
def replace_conditions(section_data, condition_variables):
            for sub_condition in section_data['sub_conditions']:
                sub_condition['condition'] = replace_condition_value_with_data(
                    sub_condition['condition'], condition_variables)
                replace_conditions(sub_condition, condition_variables)
```
and then I'd replace line 216 - 236 for a call to the previous function. Just like this:
`replace_conditions(section_data, condition_variables)`

Comment (mmumby) [reply services/core-api/app/api/now_applications/models/now_application_document_type.py]: great idea, implemented!

**Review by areyeslo — APPROVED** [review]
Very well done!!! The new functionality is awesome. Go to prod!!

**Pull request merged**
